### PR TITLE
Fix the test 02783_parsedatetimebesteffort_syslog flakiness

### PR DESCRIPTION
**Pull request by sirvickr**
Follow-up to https://github.com/ClickHouse/ClickHouse/pull/50925

### Changelog category (leave one):

- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by sirvickr** [comment]
@rschu1ze please be a reviewer of this PR.

**Comment by alexey-milovidov** [comment]
What's the point of the removal of Samoa?

**Comment by robot-clickhouse** [comment]
This is an automated comment for commit bfd3154190350d7dc0269c90062541f6936a2b84 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51112/bfd3154190350d7dc0269c90062541f6936a2b84/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Comment by sirvickr** [comment]
The `parseDateTimeBestEffort` function internally uses the current moment of time.

Firstly, I intentionally take a moment  a minute earlier and test that the result will contain the current year.
After that, I passed the argument with `Samoa` time zone in order to "move" the argument over the current point of time to the future (in terms of the current server time zone) and thus ensure that the result will moved to the previous year by the function.

The opposite occurs when I take the argument a minute ahead of the current moment and then call the function again with `Auckland` timezone to "move" the argument to the past and ensure that the result contains the current year.

However, this technique (using the timezone argument of the function) inherently implies that I rely on the current server time zone which is a bad idea as it is prone to the test's flakiness due to time zones randomization. It works perfectly well when the server timezone is later then `Samoa` and earlier than `Auckland` and fails otherwise.

In a nutshell, using time zones in the test turns out to be a mistake.

**Comment by sirvickr** [comment]
The [original PR](https://github.com/ClickHouse/ClickHouse/pull/50925) contains thorough description of how the test works and sheds lite on another strange problem with `arrayJoin()` that I had to work around previously.

**Review by alexey-milovidov — APPROVED** [review]
